src/main/scala/Joint.scala

```diff
@@ -47,14 +113,28 @@ object Joint {
   * @param shape A list of lines specifying the shape of the joint. Each item is a
   * 3-tuple. The first two items specify the line, while the last gives the distance
   * of the line from the joint's center in the local coordinate system.
+  * @param dipAngleParam An optional parameter that can be used to specify the dip angle for the joint.
+  *                 This avoids recalculation of a known dip angle.
+  * @param dipDirectionParam An optional parameter that can be used to specify the dip direction for
+  *                     the joint. This avoids recalculation of a known dip direction.
   */
 case class Joint(normalVec: (Double, Double, Double), localOrigin: (Double, Double, Double),
-                 center: (Double, Double, Double), dipAngle: Double, dipDirection: Double, phi: Double,
-                 cohesion: Double, shape: Seq[((Double, Double, Double),Double)]) {
+                 center: (Double, Double, Double), phi: Double, cohesion: Double,
+                 shape: Seq[((Double, Double, Double),Double)],
+                 dipAngleParam: Option[Double]=null,
```

Comment:
Ah, okay, that makes sense. I'll admit the Option/Some/None stuff is only slowly sinking in. I've updated it as you describe above. You'll notice in the updateJoint method, I have the inputs as `Some(dipAngle)` and `Some(dipDirection)` since the constructor requires `Option[Double]`. 

Is this okay?